src/playlist-loader.js

```diff
@@ -338,76 +340,90 @@ const PlaylistLoader = function(srcUrl, withCredentials) {
     });
   });
 
-  // request the specified URL
-  request = XhrModule({
-    uri: srcUrl,
-    withCredentials
-  }, function(error, req) {
-    let parser;
-    let playlist;
-    let i;
+  loader.pause = () => {
+    loader.stopRequest();
+    window.clearTimeout(mediaUpdateTimeout);
+  };
 
-    // clear the loader's request reference
-    request = null;
+  loader.resume = () => {
+    loader.trigger('mediaupdatetimeout');
+  };
 
-    if (error) {
-      loader.error = {
-        status: req.status,
-        message: 'HLS playlist request error at URL: ' + srcUrl,
-        responseText: req.responseText,
-        // MEDIA_ERR_NETWORK
-        code: 2
-      };
-      return loader.trigger('error');
-    }
+  loader.start = () => {
+    loader.started = true;
 
-    parser = new m3u8.Parser();
-    parser.push(req.responseText);
-    parser.end();
+    // request the specified URL
+    request = XhrModule({
+      uri: srcUrl,
+      withCredentials
+    }, function(error, req) {
+      let parser;
+      let playlist;
+      let i;
 
-    loader.state = 'HAVE_MASTER';
+      // clear the loader's request reference
+      request = null;
 
-    parser.manifest.uri = srcUrl;
+      if (error) {
+        loader.error = {
+          status: req.status,
+          message: 'HLS playlist request error at URL: ' + srcUrl,
+          responseText: req.responseText,
+          // MEDIA_ERR_NETWORK
+          code: 2
+        };
+        return loader.trigger('error');
+      }
 
-    // loaded a master playlist
-    if (parser.manifest.playlists) {
-      loader.master = parser.manifest;
+      parser = new m3u8.Parser();
```

Comment:
I think that is just the diff. What I did here was just move the initial loading into the function loader.start and changed the indentation level.